src/overmap.cpp

```diff
@@ -1348,139 +1385,95 @@ void overmap::draw(WINDOW *w, WINDOW *wbar, const tripoint &center,
         }
     }
 
-    for (int i = 0; i < om_map_width; i++) {
-        for (int j = 0; j < om_map_height; j++) {
-            const int omx = cursx + i - (om_map_width / 2);
-            const int omy = cursy + j - (om_map_height / 2);
+    // A small LRU cache: most oter_id's occur in clumps like forests of swamps.
+    // This cache helps avoid much more costly lookups in the full hashmap.
+    constexpr size_t cache_size = 8;
+    std::array<std::pair<oter_id, oter_t const*>, cache_size> cache {};
+    size_t cache_next = 0;
+
+    auto const offset_x = cursx - (om_map_width  / 2);
```

Comment:
The `cache_size` variable? I stuck that in there because of the rather larger separation between here and where `cache_size` is used again to calculate the next index. I could remove it, but if the size changes you'd have to be careful to update both locations properly.